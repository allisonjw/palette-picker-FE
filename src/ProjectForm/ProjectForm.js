import React, { useState} from 'react';
import {useDispatch } from 'react-redux';
import { addNewProject } from '../actions';
import { addProject } from '../util/apiCalls';
import './ProjectForm.scss';
export const ProjectForm = () => {
    const dispatch = useDispatch();
    const [newProjectName, setNewProjectName] = useState('');

    const handleSubmit = async (e) => {
        e.preventDefault();
        const newProject = { 
            project_name: newProjectName,  
        };
        addProject(newProject)
        dispatch(addNewProject(newProject));
        setNewProjectName('')
    };
    
    return(
        <form className="project_form" onSubmit={(e) => handleSubmit(e)}>
            <input 
                className="project_input"
                placeholder="Enter Project Name"
                value={newProjectName}
                onChange={(e) => setNewProjectName(e.target.value)}/>
            <button 
                className="project_btn">Submit Project
            </button>
        </form>
    )
}
export default ProjectForm;