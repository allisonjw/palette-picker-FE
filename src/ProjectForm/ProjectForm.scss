%form_styling {
    height: 40px;
    font-size: 1.3em;
    // box-shadow: inset 1px 1px 2px 1px black;
    padding-left: 1.5%;
    outline: none;
    margin-left: 15px;
    min-width: 15%;
}

.project_form {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100px;
}

.project_input, 
.project_btn {
    @extend %form_styling;
}

#input_project {
    height: 34px;
    border-radius: 6px;
    font-family: 'Shadows Into Light', cursive;

}

.project_btn {
    background-color: black;
    border-radius: 6px;
    color: white;
    font-family: 'Shadows Into Light', cursive;


}

.project_btn {
    &:hover {
        cursor: pointer;
        transform: scale(1.05);
        transition: all .4s ease-in-out;
    }
}
