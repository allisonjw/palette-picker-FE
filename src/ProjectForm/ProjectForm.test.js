import React from 'react';
import { render } from '@testing-library/react';
import ProjectForm from './ProjectForm';
import { shallow } from 'enzyme';

describe('ProjectForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ProjectForm />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});