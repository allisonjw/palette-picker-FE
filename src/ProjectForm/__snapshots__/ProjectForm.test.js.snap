// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm should match the snapshot 1`] = `
<form
  className="project_form"
>
  <input
    className="project_input"
  />
  <button
    className="project_btn"
  >
    Submit Project
  </button>
</form>
`;
